### DISCUSSION TITLE
Under what circumstances would I want to use this?

### DISCUSSION THREAD

**Discussion by dhnaranjo** [jeremyevans/ruby-refrigerator#9] [General]
I saw this gem linked in www.rubyweekly.com and, being a fan of Roda n Rodauth, took a look. I didn't even know core classes being unexpectedly modified was something I could be worried about. Should I be worried? Is this protecting against malice? Updates to other dependencies introducing weird monkey patches? Something else?

Google searches on the subject return this repo and red refrigerators.

**Comment by jeremyevans** [comment]
I use this gem in all of my applications that use Roda.  In my opinion, this gem is mostly useful for two main things:

1. Checking that requiring a library only modifies what you expect it to modify (that's what the `check_require` executable does).
2. Freezing all core classes in an application after the application has been setup, but before it starts accepting requests.  At that point, no core classes should be modified.  This can catch issues with poorly programmed dependencies modifying core classes at runtime.  The most common case this will catch is a library requiring something at runtime that adds a constant to `Object`.  As to whether or not you should be worried about it, I guess it depends on how paranoid you are and how much faith you have in your dependencies. :)

**Reply by dhnaranjo** [reply]
Appreciate the explanation, and your work generally. Good looking out.
